### PR TITLE
Fix NullPointer on multiple imports of unavailable modules

### LINKED ISSUE #18449: Need negative test for multiple file same package import
**Description:**
Need a negative test for same package import in multiple files. (if the package is not found), currently this throws nullpointer exception

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### LINKED ISSUE #18505: Getting a bad-sad error when there are compile errors in the project
**Description:**

Project structure:
```
project
├── Ballerina.toml
└── src
    ├── foo
    │   └── src
    │       └── foo
    │           ├── annotation.bal
    │           └── api_gateway_cache.bal
    │
    └── bar
        └── src
            └── some-directory
                ├── interceptors
                ├── listeners.bal
                │
                └── token_revocation_extension.bal
```
When the above project is built using `ballerina build` command, there were compile errors in the `foo` module. Along with the compile errors, there was a bad-sad error as well, that was logged.

ballerina-internal.log contains the following:
```
[2019-09-04 09:30:39,431] SEVERE {b7a.log.crash} - null 
java.lang.NullPointerException
	at org.wso2.ballerinalang.programfile.PackageFileWriter.writePackage(PackageFileWriter.java:51)
	at org.ballerinalang.packerina.writer.BirFileWriter.writeBIRToPath(BirFileWriter.java:100)
	at org.ballerinalang.packerina.task.CreateBirTask.writeImportBir(CreateBirTask.java:68)
	at org.ballerinalang.packerina.task.CreateBirTask.execute(CreateBirTask.java:48)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:388)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)

```

**Affected Versions:**
v1.0-pre-beta (next-release branch for beta)

**Source code:**
[api-gateway.zip](https://github.com/ballerina-platform/ballerina-lang/files/3631003/api-gateway.zip)

### PR DESCRIPTION
**Pull request by irshadnilam**
## Purpose
$title

Fixes #18505
Fixes #18449
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19224
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/imports/invalid-import/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[project]
+org-name = "invalid-import"
+version = "1.0.0"
```
Comment (rdhananjaya) [inline tests/jballerina-unit-test/src/test/resources/test-src/imports/invalid-import/Ballerina.toml]: Need a new line here.

Comment (irshadnilam) [reply tests/jballerina-unit-test/src/test/resources/test-src/imports/invalid-import/Ballerina.toml]: Fixed.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/imports/ImportsNegativeTests.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Negative test cases for imports.
+ *
+ *  @since 1.1.0
+ */
+public class ImportsNegativeTests {
+
+    @Test(description = "Test invalid import on single file")
+    public void testSelfImport() {
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/imports/ImportsNegativeTests.java]: Is the name correct?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 3]
```diff
@@ -266,7 +266,11 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
             for (BLangImportPackage unresolvedPkg : importHolder.unresolved) {
                 BPackageSymbol pkgSymbol = importHolder.resolved.symbol; // get a copy of the package symbol, add
                                                                          // compilation unit info to it,
-    
+                if (pkgSymbol == null) {
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Can't this be checked in the outer for loop? Or we are repeating logic here for each unresolvedPkg?

Comment (irshadnilam) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Moved `BPackageSymbol pkgSymbol = importHolder.resolved.symbol;` to the outer loop.  
We need that check inside the inner loop to printout the message.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/imports/ImportsNegativeTests.java (hunk 2 above)
Comment (irshadnilam) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/imports/ImportsNegativeTests.java]: Fixed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/RepoHierarchy.java [hunk 4]
```diff
@@ -53,9 +55,15 @@ public Resolution resolve(PackageID pkg) {
             Repo repo = repos[i];
             Patten patten = repo.calculate(pkg);
             if (patten != Patten.NULL) {
+                List<CompilerInput> paths = new ArrayList<>();
                 Converter converter = repo.getConverterInstance();
-                List<CompilerInput> paths = patten.convertToSources(converter, pkg)
-                                                  .collect(Collectors.toList());
+                try {
+                    paths = patten.convertToSources(converter, pkg)
+                            .collect(Collectors.toList());
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/RepoHierarchy.java]: Shall we combine these lines?

Comment (irshadnilam) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/RepoHierarchy.java]: Refactored

**Pull request merged**